playground/src/pages/CodingEnvironment.tsx

```diff
@@ -214,9 +212,33 @@ Cycles = [${trace_length.toString()}]`);
       toast.error("Not yet available");
     });
   };
+
+  /**
+   * This verifies the proof that is stored in the session. 
+   * It runs the Rust program that is imported above.
+   * It returns true if the proof is valid, and false otherwise.
+   * As inputs we need program_info, stack_input, stack_output, and proof.
+   */
   const verifyProgram = async () => {
     init().then(() => {
-      toast.error("Not yet available");
+      try {
+
+        if (checkInputs(inputs)) {
+          const result = verify_program(code, inputs, output, proof);
+          setOutput(`
+Verification success with security level of ${result}
```

Comment:
nit: We should talk about the bits of security rather than just showing a number. I would change this to "Verification succeeded with a security level of ${result} bits."

Note for the future: we should be able to easily make it so people can change the security level. I think this would be cool to do.

Reply:
very nice idea!